.github/workflows/ci-checks.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow runs on pull requests when they are opened, synchronized, or reopened.
+# It performs the following actions:
+# 1. Resets the 'ci-passed' label status when a pull request is synchronized or reopened, 
+#    indicating that changes have been pushed and CI needs to rerun.
+# 2. Polls every 5 minutes to check if all CI checks have passed.
+# 3. Adds the 'ci-passed' label to the pull request if all CI checks pass successfully.
+
+name: Check CI Completion
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  checks: read  
+  pull-requests: write
+
+jobs:
+  reset_ci_passed_label:
+    if: github.event_name == 'pull_request' && (github.event.action == 'synchronize' || github.event.action == 'reopened')
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Reset ci-passed label status on PR Syncronization
+        run: |
+          echo "Resetting 'ci-passed' label as new changes have been pushed."
+          gh pr edit ${{ github.event.pull_request.number }} --remove-label "ci-passed" --repo $GITHUB_REPOSITORY || echo "Label not present"
```

Comment:
same as https://github.com/kubeflow/pipelines/pull/11262/files#r1783673378
I believe we'd need to add a step to create the label for this repo: https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels

Reply:
@DharmitD  That's correct. @HumairAK  added the 'ci-passed' label to the repo.